packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2081,6 +2081,29 @@ typedef ElementVisitor = void Function(Element element);
 ///
 /// {@youtube 560 315 https://www.youtube.com/watch?v=rIaaH87z1-g}
 ///
+/// Avoid storing instances of [BuildContext]s because they may become invalid
+/// if the widget they are associated with is unmounted from the widget tree.
+/// {@template flutter.widgets.BuildContext.asynchronous_gap}
+/// If a [BuildContext] is used across an asynchronous gap (i.e. after performing
+/// an asynchronous operation), consider checking [mounted] to determine whether
+/// the context is still valid before interacting with it:
+///
+/// ```dart
```

Comment:
This is almost a functional example app. Would it make sense to make it into a full sample?

Reply:
Hm, it doesn't demonstrate anything visually though. In this case I think a full sample would actually distract from the point this is trying to make...

Reply:
Ok, makes sense.  I mean it could show a dialog going away after a second, but I agree it's not all that visual.